### PR TITLE
[YUNIKORN-2566] Remove AllocationAsk reference from askEvents

### PR DESCRIPTION
**Pull request by pbacsko**
### What is this PR for?
Don't keep the `AllocationAsk` object inside `askEvents`. By removing it, we can move this type out of the `objects` package.

### What type of PR is it?
* [ ] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [x] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2566

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/scheduler/objects/allocation_ask.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ func NewAllocationAsk(allocationKey string, applicationID string, allocatedResou
 		resKeyPerNode:     make(map[string]string),
 	}
 	aa.resKeyWithoutNode = reservationKeyWithoutNode(applicationID, allocationKey)
-	aa.askEvents = newAskEvents(aa, events.GetEventSystem())
+	aa.askEvents = newAskEvents(events.GetEventSystem())
 	return aa
```
Comment (chia7712) [inline pkg/scheduler/objects/allocation_ask.go:83]: It seems we can simplify the code now.
```go
func NewAllocationAsk(allocationKey string, applicationID string, allocatedResource *resources.Resource) *AllocationAsk {
	return &AllocationAsk{
		allocationKey:     allocationKey,
		applicationID:     applicationID,
		allocatedResource: allocatedResource,
		allocLog:          make(map[string]*AllocationLogEntry),
		resKeyPerNode:     make(map[string]string),
		resKeyWithoutNode: reservationKeyWithoutNode(applicationID, allocationKey),
		askEvents:         newAskEvents(events.GetEventSystem()),
	}
}
```

File: pkg/scheduler/objects/allocation_ask.go [hunk 2]
```diff
@@ -108,7 +108,7 @@ func NewAllocationAskFromSI(ask *si.AllocationAsk) *AllocationAsk {
 		return nil
 	}
 	saa.resKeyWithoutNode = reservationKeyWithoutNode(ask.ApplicationID, ask.AllocationKey)
-	saa.askEvents = newAskEvents(saa, events.GetEventSystem())
+	saa.askEvents = newAskEvents(events.GetEventSystem())
```
Comment (chia7712) [inline pkg/scheduler/objects/allocation_ask.go:111]: ditto

```go
func NewAllocationAskFromSI(ask *si.AllocationAsk) *AllocationAsk {
	// this is a safety check placeholder and task group name must be set as a combo
	// order is important as task group can be set without placeholder but not the other way around
	if ask.Placeholder && ask.TaskGroupName == "" {
		log.Log(log.SchedAllocation).Debug("ask cannot be a placeholder without a TaskGroupName",
			zap.Stringer("SI ask", ask))
		return nil
	}
	return &AllocationAsk{
		allocationKey:     ask.AllocationKey,
		allocatedResource: resources.NewResourceFromProto(ask.ResourceAsk),
		applicationID:     ask.ApplicationID,
		tags:              CloneAllocationTags(ask.Tags),
		createTime:        time.Now(),
		priority:          ask.Priority,
		placeholder:       ask.Placeholder,
		taskGroupName:     ask.TaskGroupName,
		requiredNode:      common.GetRequiredNodeFromTag(ask.Tags),
		allowPreemptSelf:  common.IsAllowPreemptSelf(ask.PreemptionPolicy),
		allowPreemptOther: common.IsAllowPreemptOther(ask.PreemptionPolicy),
		originator:        ask.Originator,
		allocLog:          make(map[string]*AllocationLogEntry),
		resKeyPerNode:     make(map[string]string),
		resKeyWithoutNode: reservationKeyWithoutNode(ask.ApplicationID, ask.AllocationKey),
		askEvents: newAskEvents(events.GetEventSystem()),
	}
}
```

**Review by chia7712 — APPROVED** [review]
@pbacsko nice patch. two comments are left. They can be addressed in another PR, so feel free to merge this PR.

**Review by chenyulin0719 — APPROVED** [review]
+1 LGTM

File: pkg/scheduler/objects/allocation_ask.go (hunk 1 above)
Comment (pbacsko) [reply pkg/scheduler/objects/allocation_ask.go:83]: Done

Comment (pbacsko) [reply pkg/scheduler/objects/allocation_ask.go:83]: Done

File: pkg/scheduler/objects/allocation_ask.go (hunk 2 above)
Comment (pbacsko) [reply pkg/scheduler/objects/allocation_ask.go]: Done

**Pull request closed without merging**